### PR TITLE
PP-10584 add settings widget docs

### PR DESCRIPTION
**Pull request by rolandkawka**
# Deploy preview

Scroll down to the checks section for the link to the deploy preview of your branch.

# Links

Link your Jira ticket.

- [Jira](https://livechatinc.atlassian.net/browse/PP-10584)

# Description

Added docs for createSettingsWidget

# Review and release

Apply changes and resolve conflicts. Once the described functionality lands on prod, let us know on #platform-docs-releases that your PR is ready for release. We will **merge** and **publish** the changes.

### CODE REVIEW

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 1]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
 
 </CodeSample>
 
+## Settings widgets
+
+If you want to connect a Settings widget to the LiveChat Agent App, you should use the `createSettingsWidget` function. It returns a promise resolving to a Settings widget instance.
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:448]:
```suggestion
If you want to connect a Settings widget to the LiveChat App, you should use the `createSettingsWidget` function. It returns a promise resolving to a Settings widget instance.
```

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 2]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+  // read the page data
+});
+```
+
+</CodeSample>
+</Code>
+
+### Methods
+
+- [Get page data](#get-page-data)
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:486]:
```suggestion
```

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 3]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+- [redirect](#redirect)
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:487]:
```suggestion
```

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 4]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+
+```js
+widget.getPageData();
+```
+
+</CodeSample>
+
+#### Redirect
+
+Redirects using the main window. Calling this method will send postmessage to Agent App, witch will be handled there. After that, redirect using window object is performed.
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:503]:
```suggestion
Redirects using the main window. Calling this method will send `postmessage` to the LiveChat App, which will be handled there. After that, a redirect using a window object is performed.
```

Comment (oliwiapolec) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:503]: what purpose does `postmessage` serve here? Shouldn't it be `postMessage` as well?

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 5]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+</Code>
+
+### Methods
+
+- [Get page data](#get-page-data)
+- [redirect](#redirect)
+
+#### Get page data
+
+Gets the main window page data recorded most recently. Returns the `IPageData` object, which is identical to the one emitted by the `page_data` event or null (if no data were registered).
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:491]:
```suggestion
Gets the most recently recorded main window page data and returns the `IPageData` object (identical to the one emitted by the `page_data` event). If no data was recorded, it returns null.
```

Comment (oliwiapolec) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:491]: also, is the `IPageData` correct?

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 6]
```diff
@@ -443,6 +443,75 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+```
+
+</CodeSample>
+</Code>
+
+### Events
+
+#### `page_data`
+
+Emitted when widget in initialized. The handler will get the main window page data object as an argument.
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:468]:
```suggestion
Emitted when the widget is initialized. The handler will get the main window page data object as an argument.
```

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 4 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:503]: It's an util allowing the iframed app communicate with parent (Agent App)
https://github.com/livechat/products-sdk/blob/67928fbdfbc20a85b98a75645464c60dabb8f381/packages/widget-core-sdk/src/connection/driver.ts#L29

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 5 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:491]: It's correct - https://github.com/livechat/products-sdk/blob/master/packages/agent-app-sdk/src/widgets/shared/page-data.ts#L3

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 1 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 9304fdfe99adbde9b902385cc7cdae59b4d4bc4d

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 2 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 9304fdfe99adbde9b902385cc7cdae59b4d4bc4d

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 4 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 9304fdfe99adbde9b902385cc7cdae59b4d4bc4d - fixed copy

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 5 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 9304fdfe99adbde9b902385cc7cdae59b4d4bc4d - fixed copy

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx (hunk 6 above)
Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 9304fdfe99adbde9b902385cc7cdae59b4d4bc4d

File: src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx [hunk 7]
```diff
@@ -443,6 +443,72 @@ widget.setReportsFilters({ tag: ['chatbot'] });
...
+```
+
+</CodeSample>
+</Code>
+
+### Methods
+
+#### Get page data
+
+Gets the most recently recorded main window page data and returns the `IPageData` object (identical to the one emitted by the `page_data` event). If no data was recorded, it returns null.
```
Comment (oliwiapolec) [inline src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:488]:
```suggestion
Gets the most recently recorded main window page data and returns the `IPageData` object (identical to the one emitted by the `page_data` event). If no data was recorded, it returns `null`.
```

Comment (oliwiapolec) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx:488]: I now noticed other nulls on the page are as snippets as well, so just this one small change and I'll merge the PR 🙏

Comment (rolandkawka) [reply src/pages/extending-agent-app/products-sdk/agent-app-sdk/index.mdx]: 116a26eb652fbf852616baa506c9451b12048e14

**Pull request merged**